pkg/operator/starter.go

```diff
@@ -354,3 +425,76 @@ func withCustomTags(infraLister v1.InfrastructureLister) deploymentcontroller.De
 		return nil
 	}
 }
+
+func assetWithNamespaceFunc(namespace string) resourceapply.AssetFunc {
+	return func(name string) ([]byte, error) {
+		content, err := assets.ReadFile(name)
+		if err != nil {
+			panic(err)
+		}
+		return bytes.ReplaceAll(content, []byte("${NAMESPACE}"), []byte(namespace)), nil
+	}
+}
+
+func withNamespaceDeploymentHook(namespace string) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		deployment.Namespace = namespace
+		return nil
+	}
+}
+
+func withHypershiftReplicasHook(isHypershift bool, guestNodeLister corev1listers.NodeLister) dc.DeploymentHookFunc {
+	if !isHypershift {
+		return csidrivercontrollerservicecontroller.WithReplicasHook(guestNodeLister)
+	}
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		// TODO: get this information from HostedControlPlane.Spec.AvailabilityPolicy
+		replicas := int32(1)
+		deployment.Spec.Replicas = &replicas
+		return nil
+	}
+
+}
+
+func withHypershiftDeploymentHook(isHypershift bool) dc.DeploymentHookFunc {
```

Comment:
How does the driver get cloud credentials in the mgmt cluster? I needed to add a new minter sidecar. See my proof of concept Deployment attached.
[controller.yaml.txt](https://github.com/openshift/aws-ebs-csi-driver-operator/files/9573900/controller.yaml.txt)

Reply:
Done!

Reply:
Thank you! Done.